tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -367,6 +395,24 @@ where
             write!(writer, "{} ", fmt_level)?;
         }
 
+        if self.display_thread_name {
+            match std::thread::current().name() {
+                Some(name) => {
+                    write!(writer, "{} ", FmtThreadName::new(name))?;
+                }
+                None => {
+                    // fall-back to thread id when name is absent and id's are not enabled
+                    if !self.display_thread_id {
+                        write!(writer, "{} ", FmtThreadId::new(std::thread::current().id()))?;
+                    }
+                }
+            }
+        }
+
+        if self.display_thread_id {
+            write!(writer, "{} ", FmtThreadId::new(std::thread::current().id()))?;
```

Comment:
I don't think the `FmtThreadId` newtype is really necessary just to add a couple characters of padding...we could just write
```suggestion
                        write!(writer, "{:>2?} ", std::thread::current().id())?;
                    }
                }
            }
        }

        if self.display_thread_id {
            write!(writer, "{:>2?} ", std::thread::current().id())?;
```
instead, which is a bit simpler...

Reply:
Yep, That makes sense! Just thought it would be better to have a single place which knows how to format threadId's :+1: